tests/pipeline/test_buyback_auth.py

```diff
@@ -0,0 +1,511 @@
+)
+from zipline.pipeline.loaders.buyback_auth import \
+    CashBuybackAuthorizationsLoader, ShareBuybackAuthorizationsLoader
+from zipline.pipeline.loaders.blaze import (
+    BlazeCashBuybackAuthorizationsLoader,
+    BlazeShareBuybackAuthorizationsLoader,
+    BUYBACK_ANNOUNCEMENT_FIELD_NAME,
+    SHARE_COUNT_FIELD_NAME,
+    SID_FIELD_NAME,
+    TS_FIELD_NAME,
+    CASH_FIELD_NAME
+)
+from zipline.utils.numpy_utils import make_datetime64D, NaTD
+from zipline.utils.test_utils import (
+    gen_calendars,
+    make_simple_equity_info,
+    num_days_in_range,
+    tmp_asset_finder,
+)
+
+
+sids = A, B, C, D, E = range(5)
+
+equity_info = make_simple_equity_info(
+    sids,
+    start_date=pd.Timestamp('2013-01-01', tz='UTC'),
+    end_date=pd.Timestamp('2015-01-01', tz='UTC'),
+)
+
+buyback_authorizations = {
+    # K1--K2--A1--A2--SC1--SC2--V1--V2.
+    A: pd.DataFrame({
+        "timestamp": pd.to_datetime(['2014-01-05', '2014-01-10']),
+        BUYBACK_ANNOUNCEMENT_FIELD_NAME: pd.to_datetime(['2014-01-15',
+                                                         '2014-01-20']),
+        SHARE_COUNT_FIELD_NAME: [1, 15],
+        CASH_FIELD_NAME: [10, 20]
+    }),
+    # K1--K2--E2--E1.
+    B: pd.DataFrame({
+        "timestamp": pd.to_datetime(['2014-01-05', '2014-01-10']),
+        BUYBACK_ANNOUNCEMENT_FIELD_NAME: pd.to_datetime([
+            '2014-01-20', '2014-01-15'
+        ]),
+        SHARE_COUNT_FIELD_NAME: [7, 13], CASH_FIELD_NAME: [10, 22]
+    }),
+    # K1--E1--K2--E2.
+    C: pd.DataFrame({
+        "timestamp": pd.to_datetime(['2014-01-05', '2014-01-15']),
+        BUYBACK_ANNOUNCEMENT_FIELD_NAME: pd.to_datetime([
+            '2014-01-10', '2014-01-20'
+        ]),
+        SHARE_COUNT_FIELD_NAME: [3, 1],
+        CASH_FIELD_NAME: [4, 7]
+    }),
+    # K1 == K2.
+    D: pd.DataFrame({
+        "timestamp": pd.to_datetime(['2014-01-05'] * 2),
+        BUYBACK_ANNOUNCEMENT_FIELD_NAME: pd.to_datetime([
+            '2014-01-10', '2014-01-15'
+        ]),
+        SHARE_COUNT_FIELD_NAME: [6, 23],
+        CASH_FIELD_NAME: [1, 2]
+    }),
+    E: pd.DataFrame(
+        columns=["timestamp",
+                 BUYBACK_ANNOUNCEMENT_FIELD_NAME,
+                 SHARE_COUNT_FIELD_NAME,
+                 CASH_FIELD_NAME],
+        dtype='datetime64[ns]'
+    ),
+}
+
+# Must be a list - can't use generator since this needs to be used more than
+# once.
+param_dates = list(gen_calendars(
+    '2014-01-01',
+    '2014-01-31',
+    critical_dates=pd.to_datetime([
+        '2014-01-05',
+        '2014-01-10',
+        '2014-01-15',
+        '2014-01-20',
+    ]),
+))
+
+
+def zip_with_floats(dates, flts):
+    return pd.Series(flts, index=dates).astype('float')
+
+
+def num_days_between(dates, start_date, end_date):
+    return num_days_in_range(dates, start_date, end_date)
+
+
+def zip_with_dates(index_dates, dts):
+    return pd.Series(pd.to_datetime(dts), index=index_dates)
+
+
+class BuybackAuthLoaderCommonTest:
```

Comment:
I read up on it, so I think I know what you mean, but could you confirm: I need to subclass from object, right?